src/Core/Grid/Presenter/GridPresenter.php

```diff
@@ -0,0 +1,134 @@
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\Presenter;
+
+use PrestaShop\PrestaShop\Core\Grid\Column\ColumnInterface;
+use PrestaShop\PrestaShop\Core\Grid\Definition\DefinitionInterface;
+use PrestaShop\PrestaShop\Core\Grid\GridInterface;
+use Symfony\Component\Form\FormFactoryInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * Class GridPresenter is responsible for presenting grid
+ */
+final class GridPresenter implements GridPresenterInterface
+{
+    /**
+     * @var FormFactoryInterface
+     */
+    private $formFactory;
+
+    /**
+     * @param FormFactoryInterface $formFactory
+     */
+    public function __construct(FormFactoryInterface $formFactory)
+    {
+        $this->formFactory = $formFactory;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function present(GridInterface $grid)
+    {
+        $definition = $grid->getDefinition();
+        $searchCriteria = $grid->getSearchCriteria();
+        $data = $grid->getData();
+
+        list(
+            $columns,
+            $filterForm
+        ) = $this->presentColumns($definition);
+
+        return [
+            'id' => $definition->getId(),
+            'name' => $definition->getName(),
+            'filter_form' => $filterForm->createView(),
+            'columns' => $columns,
+            'actions' => [
+                'grid' => $definition->getGridActions()->toArray(),
+                'bulk' => $definition->getBulkActions()->toArray(),
+            ],
+            'data' => [
+                'rows' => $data->getRows(),
+                'rows_total' => $data->getRowsTotal(),
+                'query' => $data->getQuery(),
+            ],
+            'pagination' => [
+                'offset' => $searchCriteria->getOffset(),
+                'limit' => $searchCriteria->getLimit(),
+            ],
+            'sorting' => [
+                'order_by' => $searchCriteria->getOrderBy(),
+                'order_way' => $searchCriteria->getOrderWay(),
+            ],
+        ];
+    }
+
+    /**
+     * Get presented columns with filter form
+     *
+     * @param DefinitionInterface $definition
+     *
+     * @return array
+     */
+    private function presentColumns(DefinitionInterface $definition)
+    {
+        $formBuilder = $this->formFactory->createNamedBuilder($definition->getId());
+        $columnsArray = [];
+
+        /** @var ColumnInterface $column */
+        foreach ($definition->getColumns() as $column) {
+            $resolver = new OptionsResolver();
+            $column->configureOptions($resolver);
+            $columnOptions = $resolver->resolve($column->getOptions());
+
+            $columnsArray[] = [
+                'id' => $column->getId(),
+                'name' => $column->getName(),
+                'type' => $column->getType(),
+                'options' => $columnOptions,
+            ];
+
+            if (isset(
+                $columnOptions['filter_type'],
```

Comment:
i think this pushes convention to define column filters using `filter_type` & `filter_type_options`, should this be updated? :/

Reply:
what do you mean?